File: bpsdka/bpsdka4130/Samples/Print-SDK-Demo-Android/AutoGen/BFElement/output/oc/BFElementPrintDefinition.m


#import "BFElementPrintDefinition.h"

@implementation BFElementPrintAutoCutDisableForSpecialTape
+ (nonnull NSString*)key {
  return @"PrintAutoCutDisableForSpecialTape";
}
@end

@implementation BFElementPrintAutoCutEvery
+ (nonnull NSString*)key {
  return @"PrintAutoCutEvery";
}
@end

@implementation BFElementPrintAutoCutEveryCyclePerPage
+ (nonnull NSString*)key {
  return @"PrintAutoCutEveryCyclePerPage";
}
@end

@implementation BFElementPrintAutoCutLast
+ (nonnull NSString*)key {
  return @"PrintAutoCutLast";
}
@end

@implementation BFElementPrintAutoHalfCut
+ (nonnull NSString*)key {
  return @"PrintAutoHalfCut";
}
@end

@implementation BFElementPrintBidirectionalCommunication
+ (nonnull NSString*)key {
  return @"PrintBidirectionalCommunication";
}
@end

@implementation BFElementPrintCheckMediaBackgroundColor
+ (nonnull NSString*)key {
  return @"PrintCheckMediaBackgroundColor";
}
@end

@implementation BFElementPrintCheckMediaTextColor
+ (nonnull NSString*)key {
  return @"PrintCheckMediaTextColor";
}
@end

@implementation BFElementPrintCollate
+ (nonnull NSString*)key {
  return @"PrintCollate";
}
@end

@implementation BFElementPrintCopies
+ (nonnull NSString*)key {
  return @"PrintCopies";
}
@end

@implementation BFElementPrintCutPause
+ (nonnull NSString*)key {
  return @"PrintCutPause";
}
@end

@implementation BFElementPrintDashLine
+ (nonnull NSString*)key {
  return @"PrintDashLine";
}
@end

NSString * __nonnull const BFElementPrintDensityStringWeakLevel5 = @"WeakLevel5";
NSString * __nonnull const BFElementPrintDensityStringWeakLevel4 = @"WeakLevel4";
NSString * __nonnull const BFElementPrintDensityStringWeakLevel3 = @"WeakLevel3";
NSString * __nonnull const BFElementPrintDensityStringWeakLevel2 = @"WeakLevel2";
NSString * __nonnull const BFElementPrintDensityStringWeakLevel1 = @"WeakLevel1";
NSString * __nonnull const BFElementPrintDensityStringNeutral = @"Neutral";
NSString * __nonnull const BFElementPrintDensityStringStrongLevel1 = @"StrongLevel1";
NSString * __nonnull const BFElementPrintDensityStringStrongLevel2 = @"StrongLevel2";
NSString * __nonnull const BFElementPrintDensityStringStrongLevel3 = @"StrongLevel3";
NSString * __nonnull const BFElementPrintDensityStringStrongLevel4 = @"StrongLevel4";
NSString * __nonnull const BFElementPrintDensityStringStrongLevel5 = @"StrongLevel5";
NSString * __nonnull const BFElementPrintDensityStringUsePrinterSetting = @"UsePrinterSetting";
NSString * __nonnull const BFElementPrintDensityString__Null = @"__Null";
@implementation BFElementPrintDensity
+ (nonnull NSString*)convertToString:(BFElementPrintDensityNumber)num {
  switch(num) {
  case BFElementPrintDensityNumberWeakLevel5: return BFElementPrintDensityStringWeakLevel5;
  case BFElementPrintDensityNumberWeakLevel4: return BFElementPrintDensityStringWeakLevel4;
  case BFElementPrintDensityNumberWeakLevel3: return BFElementPrintDensityStringWeakLevel3;
  case BFElementPrintDensityNumberWeakLevel2: return BFElementPrintDensityStringWeakLevel2;
  case BFElementPrintDensityNumberWeakLevel1: return BFElementPrintDensityStringWeakLevel1;
  case BFElementPrintDensityNumberNeutral: return BFElementPrintDensityStringNeutral;
  case BFElementPrintDensityNumberStrongLevel1: return BFElementPrintDensityStringStrongLevel1;
  case BFElementPrintDensityNumberStrongLevel2: return BFElementPrintDensityStringStrongLevel2;
  case BFElementPrintDensityNumberStrongLevel3: return BFElementPrintDensityStringStrongLevel3;
  case BFElementPrintDensityNumberStrongLevel4: return BFElementPrintDensityStringStrongLevel4;
  case BFElementPrintDensityNumberStrongLevel5: return BFElementPrintDensityStringStrongLevel5;
  case BFElementPrintDensityNumberUsePrinterSetting: return BFElementPrintDensityStringUsePrinterSetting;
  case BFElementPrintDensityNumber__Null: return BFElementPrintDensityString__Null;
  }
  return BFElementPrintDensityString__Null;
}
+ (BFElementPrintDensityNumber)convertToNumber:(nonnull NSString*)str {
  if ([str isEqualToString:BFElementPrintDensityStringWeakLevel5]) {
    return BFElementPrintDensityNumberWeakLevel5;
  }
  if ([str isEqualToString:BFElementPrintDensityStringWeakLevel4]) {
    return BFElementPrintDensityNumberWeakLevel4;
  }
  if ([str isEqualToString:BFElementPrintDensityStringWeakLevel3]) {
    return BFElementPrintDensityNumberWeakLevel3;
  }
  if ([str isEqualToString:BFElementPrintDensityStringWeakLevel2]) {
    return BFElementPrintDensityNumberWeakLevel2;
  }
  if ([str isEqualToString:BFElementPrintDensityStringWeakLevel1]) {
    return BFElementPrintDensityNumberWeakLevel1;
  }
  if ([str isEqualToString:BFElementPrintDensityStringNeutral]) {
    return BFElementPrintDensityNumberNeutral;
  }
  if ([str isEqualToString:BFElementPrintDensityStringStrongLevel1]) {
    return BFElementPrintDensityNumberStrongLevel1;
  }
  if ([str isEqualToString:BFElementPrintDensityStringStrongLevel2]) {
    return BFElementPrintDensityNumberStrongLevel2;
  }
  if ([str isEqualToString:BFElementPrintDensityStringStrongLevel3]) {
    return BFElementPrintDensityNumberStrongLevel3;
  }
  if ([str isEqualToString:BFElementPrintDensityStringStrongLevel4]) {
    return BFElementPrintDensityNumberStrongLevel4;
  }
  if ([str isEqualToString:BFElementPrintDensityStringStrongLevel5]) {
    return BFElementPrintDensityNumberStrongLevel5;
  }
  if ([str isEqualToString:BFElementPrintDensityStringUsePrinterSetting]) {
    return BFElementPrintDensityNumberUsePrinterSetting;
  }
  if ([str isEqualToString:BFElementPrintDensityString__Null]) {
    return BFElementPrintDensityNumber__Null;
  }
  return BFElementPrintDensityNumber__Null;
}
+ (nonnull NSString*)key {
  return @"PrintDensity";
}
@end

@implementation BFElementPrintExtraFeedDots
+ (nonnull NSString*)key {
  return @"PrintExtraFeedDots";
}
@end

NSString * __nonnull const BFElementPrintFeedModeStringNoFeed = @"NoFeed";
NSString * __nonnull const BFElementPrintFeedModeStringFixedPage = @"FixedPage";
NSString * __nonnull const BFElementPrintFeedModeStringEndOfPage = @"EndOfPage";
NSString * __nonnull const BFElementPrintFeedModeStringEndOfPageRetract = @"EndOfPageRetract";
NSString * __nonnull const BFElementPrintFeedModeString__Null = @"__Null";
@implementation BFElementPrintFeedMode
+ (nonnull NSString*)convertToString:(BFElementPrintFeedModeNumber)num {
  switch(num) {
  case BFElementPrintFeedModeNumberNoFeed: return BFElementPrintFeedModeStringNoFeed;
  case BFElementPrintFeedModeNumberFixedPage: return BFElementPrintFeedModeStringFixedPage;
  case BFElementPrintFeedModeNumberEndOfPage: return BFElementPrintFeedModeStringEndOfPage;
  case BFElementPrintFeedModeNumberEndOfPageRetract: return BFElementPrintFeedModeStringEndOfPageRetract;
  case BFElementPrintFeedModeNumber__Null: return BFElementPrintFeedModeString__Null;
  }
  return BFElementPrintFeedModeString__Null;
}
+ (BFElementPrintFeedModeNumber)convertToNumber:(nonnull NSString*)str {
  if ([str isEqualToString:BFElementPrintFeedModeStringNoFeed]) {
    return BFElementPrintFeedModeNumberNoFeed;
  }
  if ([str isEqualToString:BFElementPrintFeedModeStringFixedPage]) {
    return BFElementPrintFeedModeNumberFixedPage;
  }
  if ([str isEqualToString:BFElementPrintFeedModeStringEndOfPage]) {
    return BFElementPrintFeedModeNumberEndOfPage;
  }
  if ([str isEqualToString:BFElementPrintFeedModeStringEndOfPageRetract]) {
    return BFElementPrintFeedModeNumberEndOfPageRetract;
  }
  if ([str isEqualToString:BFElementPrintFeedModeString__Null]) {
    return BFElementPrintFeedModeNumber__Null;
  }
  return BFElementPrintFeedModeNumber__Null;
}
+ (nonnull NSString*)key {
  return @"PrintFeedMode";
}
@end

NSString * __nonnull const BFElementPrintHorizontalAlignStringLeft = @"Left";
NSString * __nonnull const BFElementPrintHorizontalAlignStringCenter = @"Center";
NSString * __nonnull const BFElementPrintHorizontalAlignStringRight = @"Right";
NSString * __nonnull const BFElementPrintHorizontalAlignString__Null = @"__Null";
@implementation BFElementPrintHorizontalAlign
+ (nonnull NSString*)convertToString:(BFElementPrintHorizontalAlignNumber)num {
  switch(num) {
  case BFElementPrintHorizontalAlignNumberLeft: return BFElementPrintHorizontalAlignStringLeft;
  case BFElementPrintHorizontalAlignNumberCenter: return BFElementPrintHorizontalAlignStringCenter;
  case BFElementPrintHorizontalAlignNumberRight: return BFElementPrintHorizontalAlignStringRight;
  case BFElementPrintHorizontalAlignNumber__Null: return BFElementPrintHorizontalAlignString__Null;
  }
  return BFElementPrintHorizontalAlignString__Null;
}
+ (BFElementPrintHorizontalAlignNumber)convertToNumber:(nonnull NSString*)str {
  if ([str isEqualToString:BFElementPrintHorizontalAlignStringLeft]) {
    return BFElementPrintHorizontalAlignNumberLeft;
  }
  if ([str isEqualToString:BFElementPrintHorizontalAlignStringCenter]) {
    return BFElementPrintHorizontalAlignNumberCenter;
  }
  if ([str isEqualToString:BFElementPrintHorizontalAlignStringRight]) {
    return BFElementPrintHorizontalAlignNumberRight;
  }
  if ([str isEqualToString:BFElementPrintHorizontalAlignString__Null]) {
    return BFElementPrintHorizontalAlignNumber__Null;
  }
  return BFElementPrintHorizontalAlignNumber__Null;
}
+ (nonnull NSString*)key {
  return @"PrintHorizontalAlign";
}
@end

@implementation BFElementPrintLeftMargin
+ (nonnull NSString*)key {
  return @"PrintLeftMargin";
}
@end

NSString * __nonnull const BFElementPrintPolicyOfProcessingPrintDataStringBatchPrint = @"BatchPrint";
NSString * __nonnull const BFElementPrintPolicyOfProcessingPrintDataStringRealTimePrint = @"RealTimePrint";
NSString * __nonnull const BFElementPrintPolicyOfProcessingPrintDataString__Null = @"__Null";
@implementation BFElementPrintPolicyOfProcessingPrintData
+ (nonnull NSString*)convertToString:(BFElementPrintPolicyOfProcessingPrintDataNumber)num {
  switch(num) {
  case BFElementPrintPolicyOfProcessingPrintDataNumberBatchPrint: return BFElementPrintPolicyOfProcessingPrintDataStringBatchPrint;
  case BFElementPrintPolicyOfProcessingPrintDataNumberRealTimePrint: return BFElementPrintPolicyOfProcessingPrintDataStringRealTimePrint;
  case BFElementPrintPolicyOfProcessingPrintDataNumber__Null: return BFElementPrintPolicyOfProcessingPrintDataString__Null;
  }
  return BFElementPrintPolicyOfProcessingPrintDataString__Null;
}
+ (BFElementPrintPolicyOfProcessingPrintDataNumber)convertToNumber:(nonnull NSString*)str {
  if ([str isEqualToString:BFElementPrintPolicyOfProcessingPrintDataStringBatchPrint]) {
    return BFElementPrintPolicyOfProcessingPrintDataNumberBatchPrint;
  }
  if ([str isEqualToString:BFElementPrintPolicyOfProcessingPrintDataStringRealTimePrint]) {
    return BFElementPrintPolicyOfProcessingPrintDataNumberRealTimePrint;
  }
  if ([str isEqualToString:BFElementPrintPolicyOfProcessingPrintDataString__Null]) {
    return BFElementPrintPolicyOfProcessingPrintDataNumber__Null;
  }
  return BFElementPrintPolicyOfProcessingPrintDataNumber__Null;
}
+ (nonnull NSString*)key {
  return @"PrintPolicyOfProcessingPrintData";
}
@end

NSString * __nonnull const BFElementPrintQualityStringLowResolutionHighSpeed = @"LowResolutionHighSpeed";
NSString * __nonnull const BFElementPrintQualityStringNormalResolutionNormalSpeed = @"NormalResolutionNormalSpeed";
NSString * __nonnull const BFElementPrintQualityStringNormalResolutionDoubleSpeed = @"NormalResolutionDoubleSpeed";
NSString * __nonnull const BFElementPrintQualityStringHighResolutionLowSpeed = @"HighResolutionLowSpeed";
NSString * __nonnull const BFElementPrintQualityStringHighResolutionUsingHRImageLowSpeed = @"HighResolutionUsingHRImageLowSpeed";
NSString * __nonnull const BFElementPrintQualityString__Null = @"__Null";
@implementation BFElementPrintQuality
+ (nonnull NSString*)convertToString:(BFElementPrintQualityNumber)num {
  switch(num) {
  case BFElementPrintQualityNumberLowResolutionHighSpeed: return BFElementPrintQualityStringLowResolutionHighSpeed;
  case BFElementPrintQualityNumberNormalResolutionNormalSpeed: return BFElementPrintQualityStringNormalResolutionNormalSpeed;
  case BFElementPrintQualityNumberNormalResolutionDoubleSpeed: return BFElementPrintQualityStringNormalResolutionDoubleSpeed;
  case BFElementPrintQualityNumberHighResolutionLowSpeed: return BFElementPrintQualityStringHighResolutionLowSpeed;
  case BFElementPrintQualityNumberHighResolutionUsingHRImageLowSpeed: return BFElementPrintQualityStringHighResolutionUsingHRImageLowSpeed;
  case BFElementPrintQualityNumber__Null: return BFElementPrintQualityString__Null;
  }
  return BFElementPrintQualityString__Null;
}
+ (BFElementPrintQualityNumber)convertToNumber:(nonnull NSString*)str {
  if ([str isEqualToString:BFElementPrintQualityStringLowResolutionHighSpeed]) {
    return BFElementPrintQualityNumberLowResolutionHighSpeed;
  }
  if ([str isEqualToString:BFElementPrintQualityStringNormalResolutionNormalSpeed]) {
    return BFElementPrintQualityNumberNormalResolutionNormalSpeed;
  }
  if ([str isEqualToString:BFElementPrintQualityStringNormalResolutionDoubleSpeed]) {
    return BFElementPrintQualityNumberNormalResolutionDoubleSpeed;
  }
  if ([str isEqualToString:BFElementPrintQualityStringHighResolutionLowSpeed]) {
    return BFElementPrintQualityNumberHighResolutionLowSpeed;
  }
  if ([str isEqualToString:BFElementPrintQualityStringHighResolutionUsingHRImageLowSpeed]) {
    return BFElementPrintQualityNumberHighResolutionUsingHRImageLowSpeed;
  }
  if ([str isEqualToString:BFElementPrintQualityString__Null]) {
    return BFElementPrintQualityNumber__Null;
  }
  return BFElementPrintQualityNumber__Null;
}
+ (nonnull NSString*)key {
  return @"PrintQuality";
}
@end

NSString * __nonnull const BFElementPrintScalingModeStringOriginal = @"Original";
NSString * __nonnull const BFElementPrintScalingModeStringFitToPrintArea = @"FitToPrintArea";
NSString * __nonnull const BFElementPrintScalingModeStringFitToPaper = @"FitToPaper";
NSString * __nonnull const BFElementPrintScalingModeStringCustom = @"Custom";
NSString * __nonnull const BFElementPrintScalingModeString__Null = @"__Null";
@implementation BFElementPrintScalingMode
+ (nonnull NSString*)convertToString:(BFElementPrintScalingModeNumber)num {
  switch(num) {
  case BFElementPrintScalingModeNumberOriginal: return BFElementPrintScalingModeStringOriginal;
  case BFElementPrintScalingModeNumberFitToPrintArea: return BFElementPrintScalingModeStringFitToPrintArea;
  case BFElementPrintScalingModeNumberFitToPaper: return BFElementPrintScalingModeStringFitToPaper;
  case BFElementPrintScalingModeNumberCustom: return BFElementPrintScalingModeStringCustom;
  case BFElementPrintScalingModeNumber__Null: return BFElementPrintScalingModeString__Null;
  }
  return BFElementPrintScalingModeString__Null;
}
+ (BFElementPrintScalingModeNumber)convertToNumber:(nonnull NSString*)str {
  if ([str isEqualToString:BFElementPrintScalingModeStringOriginal]) {
    return BFElementPrintScalingModeNumberOriginal;
  }
  if ([str isEqualToString:BFElementPrintScalingModeStringFitToPrintArea]) {
    return BFElementPrintScalingModeNumberFitToPrintArea;
  }
  if ([str isEqualToString:BFElementPrintScalingModeStringFitToPaper]) {
    return BFElementPrintScalingModeNumberFitToPaper;
  }
  if ([str isEqualToString:BFElementPrintScalingModeStringCustom]) {
    return BFElementPrintScalingModeNumberCustom;
  }
  if ([str isEqualToString:BFElementPrintScalingModeString__Null]) {
    return BFElementPrintScalingModeNumber__Null;
  }
  return BFElementPrintScalingModeNumber__Null;
}
+ (nonnull NSString*)key {
  return @"PrintScalingMode";
}
@end

@implementation BFElementPrintScalingValue
+ (nonnull NSString*)key {
  return @"PrintScalingValue";
}
@end

NSString * __nonnull const BFElementPrintSpeedStringHighSpeed = @"HighSpeed";
NSString * __nonnull const BFElementPrintSpeedStringMediumHighSpeed = @"MediumHighSpeed";
NSString * __nonnull const BFElementPrintSpeedStringMediumLowSpeed = @"MediumLowSpeed";
NSString * __nonnull const BFElementPrintSpeedStringLowSpeed = @"LowSpeed";
NSString * __nonnull const BFElementPrintSpeedStringFast_DraftQuality = @"Fast_DraftQuality";
NSString * __nonnull const BFElementPrintSpeedStringFast_LineConversion = @"Fast_LineConversion";
NSString * __nonnull const BFElementPrintSpeedStringUsePrinterSetting = @"UsePrinterSetting";
NSString * __nonnull const BFElementPrintSpeedString__Null = @"__Null";
@implementation BFElementPrintSpeed
+ (nonnull NSString*)convertToString:(BFElementPrintSpeedNumber)num {
  switch(num) {
  case BFElementPrintSpeedNumberHighSpeed: return BFElementPrintSpeedStringHighSpeed;
  case BFElementPrintSpeedNumberMediumHighSpeed: return BFElementPrintSpeedStringMediumHighSpeed;
  case BFElementPrintSpeedNumberMediumLowSpeed: return BFElementPrintSpeedStringMediumLowSpeed;
  case BFElementPrintSpeedNumberLowSpeed: return BFElementPrintSpeedStringLowSpeed;
  case BFElementPrintSpeedNumberFast_DraftQuality: return BFElementPrintSpeedStringFast_DraftQuality;
  case BFElementPrintSpeedNumberFast_LineConversion: return BFElementPrintSpeedStringFast_LineConversion;
  case BFElementPrintSpeedNumberUsePrinterSetting: return BFElementPrintSpeedStringUsePrinterSetting;
  case BFElementPrintSpeedNumber__Null: return BFElementPrintSpeedString__Null;
  }
  return BFElementPrintSpeedString__Null;
}
+ (BFElementPrintSpeedNumber)convertToNumber:(nonnull NSString*)str {
  if ([str isEqualToString:BFElementPrintSpeedStringHighSpeed]) {
    return BFElementPrintSpeedNumberHighSpeed;
  }
  if ([str isEqualToString:BFElementPrintSpeedStringMediumHighSpeed]) {
    return BFElementPrintSpeedNumberMediumHighSpeed;
  }
  if ([str isEqualToString:BFElementPrintSpeedStringMediumLowSpeed]) {
    return BFElementPrintSpeedNumberMediumLowSpeed;
  }
  if ([str isEqualToString:BFElementPrintSpeedStringLowSpeed]) {
    return BFElementPrintSpeedNumberLowSpeed;
  }
  if ([str isEqualToString:BFElementPrintSpeedStringFast_DraftQuality]) {
    return BFElementPrintSpeedNumberFast_DraftQuality;
  }
  if ([str isEqualToString:BFElementPrintSpeedStringFast_LineConversion]) {
    return BFElementPrintSpeedNumberFast_LineConversion;
  }
  if ([str isEqualToString:BFElementPrintSpeedStringUsePrinterSetting]) {
    return BFElementPrintSpeedNumberUsePrinterSetting;
  }
  if ([str isEqualToString:BFElementPrintSpeedString__Null]) {
    return BFElementPrintSpeedNumber__Null;
  }
  return BFElementPrintSpeedNumber__Null;
}
+ (nonnull NSString*)key {
  return @"PrintSpeed";
}
@end

@implementation BFElementPrintTopMargin
+ (nonnull NSString*)key {
  return @"PrintTopMargin";
}
@end

@implementation BFElementPrintUsingCarbonCopyPaper
+ (nonnull NSString*)key {
  return @"PrintUsingCarbonCopyPaper";
}
@end

NSString * __nonnull const BFElementPrintVerticalAlignStringTop = @"Top";
NSString * __nonnull const BFElementPrintVerticalAlignStringMiddle = @"Middle";
NSString * __nonnull const BFElementPrintVerticalAlignStringBottom = @"Bottom";
NSString * __nonnull const BFElementPrintVerticalAlignString__Null = @"__Null";
@implementation BFElementPrintVerticalAlign
+ (nonnull NSString*)convertToString:(BFElementPrintVerticalAlignNumber)num {
  switch(num) {
  case BFElementPrintVerticalAlignNumberTop: return BFElementPrintVerticalAlignStringTop;
  case BFElementPrintVerticalAlignNumberMiddle: return BFElementPrintVerticalAlignStringMiddle;
  case BFElementPrintVerticalAlignNumberBottom: return BFElementPrintVerticalAlignStringBottom;
  case BFElementPrintVerticalAlignNumber__Null: return BFElementPrintVerticalAlignString__Null;
  }
  return BFElementPrintVerticalAlignString__Null;
}
+ (BFElementPrintVerticalAlignNumber)convertToNumber:(nonnull NSString*)str {
  if ([str isEqualToString:BFElementPrintVerticalAlignStringTop]) {
    return BFElementPrintVerticalAlignNumberTop;
  }
  if ([str isEqualToString:BFElementPrintVerticalAlignStringMiddle]) {
    return BFElementPrintVerticalAlignNumberMiddle;
  }
  if ([str isEqualToString:BFElementPrintVerticalAlignStringBottom]) {
    return BFElementPrintVerticalAlignNumberBottom;
  }
  if ([str isEqualToString:BFElementPrintVerticalAlignString__Null]) {
    return BFElementPrintVerticalAlignNumber__Null;
  }
  return BFElementPrintVerticalAlignNumber__Null;
}
+ (nonnull NSString*)key {
  return @"PrintVerticalAlign";
}
@end

File: bpsdka/bpsdka4130/Samples/Print-SDK-Demo-Android/AutoGen/BFElement/output/cpp/bf_element_print_definition.cpp

#include "bf_element_print_definition.hpp"
#include <string>
#include <cstring>

namespace bf {
  namespace element {
    const char* PrintDensity::convertToString(Number num) {
      switch(num) {
        case Number::WeakLevel5: return String::WeakLevel5;
        case Number::WeakLevel4: return String::WeakLevel4;
        case Number::WeakLevel3: return String::WeakLevel3;
        case Number::WeakLevel2: return String::WeakLevel2;
        case Number::WeakLevel1: return String::WeakLevel1;
        case Number::Neutral: return String::Neutral;
        case Number::StrongLevel1: return String::StrongLevel1;
        case Number::StrongLevel2: return String::StrongLevel2;
        case Number::StrongLevel3: return String::StrongLevel3;
        case Number::StrongLevel4: return String::StrongLevel4;
        case Number::StrongLevel5: return String::StrongLevel5;
        case Number::UsePrinterSetting: return String::UsePrinterSetting;
        case Number::__Null: return String::__Null;
      }
      return String::__Null;
    };
    PrintDensity::Number PrintDensity::convertToNumber(const char* string) {
      if (!strcmp(string, String::WeakLevel5)) {
        return Number::WeakLevel5;
      }
      if (!strcmp(string, String::WeakLevel4)) {
        return Number::WeakLevel4;
      }
      if (!strcmp(string, String::WeakLevel3)) {
        return Number::WeakLevel3;
      }
      if (!strcmp(string, String::WeakLevel2)) {
        return Number::WeakLevel2;
      }
      if (!strcmp(string, String::WeakLevel1)) {
        return Number::WeakLevel1;
      }
      if (!strcmp(string, String::Neutral)) {
        return Number::Neutral;
      }
      if (!strcmp(string, String::StrongLevel1)) {
        return Number::StrongLevel1;
      }
      if (!strcmp(string, String::StrongLevel2)) {
        return Number::StrongLevel2;
      }
      if (!strcmp(string, String::StrongLevel3)) {
        return Number::StrongLevel3;
      }
      if (!strcmp(string, String::StrongLevel4)) {
        return Number::StrongLevel4;
      }
      if (!strcmp(string, String::StrongLevel5)) {
        return Number::StrongLevel5;
      }
      if (!strcmp(string, String::UsePrinterSetting)) {
        return Number::UsePrinterSetting;
      }
      if (!strcmp(string, String::__Null)) {
        return Number::__Null;
      }
      return Number::__Null;
    };

    const char* PrintFeedMode::convertToString(Number num) {
      switch(num) {
        case Number::NoFeed: return String::NoFeed;
        case Number::FixedPage: return String::FixedPage;
        case Number::EndOfPage: return String::EndOfPage;
        case Number::EndOfPageRetract: return String::EndOfPageRetract;
        case Number::__Null: return String::__Null;
      }
      return String::__Null;
    };
    PrintFeedMode::Number PrintFeedMode::convertToNumber(const char* string) {
      if (!strcmp(string, String::NoFeed)) {
        return Number::NoFeed;
      }
      if (!strcmp(string, String::FixedPage)) {
        return Number::FixedPage;
      }
      if (!strcmp(string, String::EndOfPage)) {
        return Number::EndOfPage;
      }
      if (!strcmp(string, String::EndOfPageRetract)) {
        return Number::EndOfPageRetract;
      }
      if (!strcmp(string, String::__Null)) {
        return Number::__Null;
      }
      return Number::__Null;
    };

    const char* PrintHorizontalAlign::convertToString(Number num) {
      switch(num) {
        case Number::Left: return String::Left;
        case Number::Center: return String::Center;
        case Number::Right: return String::Right;
        case Number::__Null: return String::__Null;
      }
      return String::__Null;
    };
    PrintHorizontalAlign::Number PrintHorizontalAlign::convertToNumber(const char* string) {
      if (!strcmp(string, String::Left)) {
        return Number::Left;
      }
      if (!strcmp(string, String::Center)) {
        return Number::Center;
      }
      if (!strcmp(string, String::Right)) {
        return Number::Right;
      }
      if (!strcmp(string, String::__Null)) {
        return Number::__Null;
      }
      return Number::__Null;
    };

    const char* PrintPolicyOfProcessingPrintData::convertToString(Number num) {
      switch(num) {
        case Number::BatchPrint: return String::BatchPrint;
        case Number::RealTimePrint: return String::RealTimePrint;
        case Number::__Null: return String::__Null;
      }
      return String::__Null;
    };
    PrintPolicyOfProcessingPrintData::Number PrintPolicyOfProcessingPrintData::convertToNumber(const char* string) {
      if (!strcmp(string, String::BatchPrint)) {
        return Number::BatchPrint;
      }
      if (!strcmp(string, String::RealTimePrint)) {
        return Number::RealTimePrint;
      }
      if (!strcmp(string, String::__Null)) {
        return Number::__Null;
      }
      return Number::__Null;
    };

    const char* PrintQuality::convertToString(Number num) {
      switch(num) {
        case Number::LowResolutionHighSpeed: return String::LowResolutionHighSpeed;
        case Number::NormalResolutionNormalSpeed: return String::NormalResolutionNormalSpeed;
        case Number::NormalResolutionDoubleSpeed: return String::NormalResolutionDoubleSpeed;
        case Number::HighResolutionLowSpeed: return String::HighResolutionLowSpeed;
        case Number::HighResolutionUsingHRImageLowSpeed: return String::HighResolutionUsingHRImageLowSpeed;
        case Number::__Null: return String::__Null;
      }
      return String::__Null;
    };
    PrintQuality::Number PrintQuality::convertToNumber(const char* string) {
      if (!strcmp(string, String::LowResolutionHighSpeed)) {
        return Number::LowResolutionHighSpeed;
      }
      if (!strcmp(string, String::NormalResolutionNormalSpeed)) {
        return Number::NormalResolutionNormalSpeed;
      }
      if (!strcmp(string, String::NormalResolutionDoubleSpeed)) {
        return Number::NormalResolutionDoubleSpeed;
      }
      if (!strcmp(string, String::HighResolutionLowSpeed)) {
        return Number::HighResolutionLowSpeed;
      }
      if (!strcmp(string, String::HighResolutionUsingHRImageLowSpeed)) {
        return Number::HighResolutionUsingHRImageLowSpeed;
      }
      if (!strcmp(string, String::__Null)) {
        return Number::__Null;
      }
      return Number::__Null;
    };

    const char* PrintScalingMode::convertToString(Number num) {
      switch(num) {
        case Number::Original: return String::Original;
        case Number::FitToPrintArea: return String::FitToPrintArea;
        case Number::FitToPaper: return String::FitToPaper;
        case Number::Custom: return String::Custom;
        case Number::__Null: return String::__Null;
      }
      return String::__Null;
    };
    PrintScalingMode::Number PrintScalingMode::convertToNumber(const char* string) {
      if (!strcmp(string, String::Original)) {
        return Number::Original;
      }
      if (!strcmp(string, String::FitToPrintArea)) {
        return Number::FitToPrintArea;
      }
      if (!strcmp(string, String::FitToPaper)) {
        return Number::FitToPaper;
      }
      if (!strcmp(string, String::Custom)) {
        return Number::Custom;
      }
      if (!strcmp(string, String::__Null)) {
        return Number::__Null;
      }
      return Number::__Null;
    };

    const char* PrintSpeed::convertToString(Number num) {
      switch(num) {
        case Number::HighSpeed: return String::HighSpeed;
        case Number::MediumHighSpeed: return String::MediumHighSpeed;
        case Number::MediumLowSpeed: return String::MediumLowSpeed;
        case Number::LowSpeed: return String::LowSpeed;
        case Number::Fast_DraftQuality: return String::Fast_DraftQuality;
        case Number::Fast_LineConversion: return String::Fast_LineConversion;
        case Number::UsePrinterSetting: return String::UsePrinterSetting;
        case Number::__Null: return String::__Null;
      }
      return String::__Null;
    };
    PrintSpeed::Number PrintSpeed::convertToNumber(const char* string) {
      if (!strcmp(string, String::HighSpeed)) {
        return Number::HighSpeed;
      }
      if (!strcmp(string, String::MediumHighSpeed)) {
        return Number::MediumHighSpeed;
      }
      if (!strcmp(string, String::MediumLowSpeed)) {
        return Number::MediumLowSpeed;
      }
      if (!strcmp(string, String::LowSpeed)) {
        return Number::LowSpeed;
      }
      if (!strcmp(string, String::Fast_DraftQuality)) {
        return Number::Fast_DraftQuality;
      }
      if (!strcmp(string, String::Fast_LineConversion)) {
        return Number::Fast_LineConversion;
      }
      if (!strcmp(string, String::UsePrinterSetting)) {
        return Number::UsePrinterSetting;
      }
      if (!strcmp(string, String::__Null)) {
        return Number::__Null;
      }
      return Number::__Null;
    };

    const char* PrintVerticalAlign::convertToString(Number num) {
      switch(num) {
        case Number::Top: return String::Top;
        case Number::Middle: return String::Middle;
        case Number::Bottom: return String::Bottom;
        case Number::__Null: return String::__Null;
      }
      return String::__Null;
    };
    PrintVerticalAlign::Number PrintVerticalAlign::convertToNumber(const char* string) {
      if (!strcmp(string, String::Top)) {
        return Number::Top;
      }
      if (!strcmp(string, String::Middle)) {
        return Number::Middle;
      }
      if (!strcmp(string, String::Bottom)) {
        return Number::Bottom;
      }
      if (!strcmp(string, String::__Null)) {
        return Number::__Null;
      }
      return Number::__Null;
    };

  }
}